### ISSUE TITLE
own dataset train

### ISSUE THREAD

**Issue by cama36** [octree-nn/octformer#23]
Hi, I'm working with my own dataset (in keeping with the sunrgbd dataset format) and I'm noticing a strange thing, everything except cls_loss is 0. It seems that the voxel_size and octree_depth and self.scale_factor = 2 / (2**octree_depth * voxel_size ) These hyperparameters don't match? Sorry I can't understand it well, thank you very much for your help, all the best!

**Comment by wang-ps** [comment]
Suppose the input point cloud is in [-10m, 10m]. 
If the `voxel_size` is set to 0.01m, then after multiplying the point cloud with `1/voxel_size`, the point cloud is scaled to [-100, 100]. 
We know that an octree with depth 8 can cover the range [-128, 128], so we can set the `octree_depth` to 8. 
For consistency, I require the point cloud to be in [-1, 1] when building octrees by calling the function `octree.build`. So we need to scale the point with  `2 / 2**octree_depth` to normalize it from [-100, 100] into [-1, 1].

In summary, we set the scale factor to `2 / (2**octree_depth * voxel_size )`.

**Comment by cama36** [comment]
I'm sorry Professor, I'm not sure I'm understanding you correctly. What you mean is that if the point cloud x,y,z ranges from (-10, 10), then 1000*2 voxels are needed to be filled in case the voxel_size is 0.01, and an octree with a depth of 10 is just enough for this resolution, so the depth needs to be greater than 10 in this case (you set it to 12), right?

By the way, my data x,y range in (0,1000), z range in (0,500), voxel_size set to 1.0, depth>=10 is this understanding correct?

Translated with DeepL.com (free version)

**Comment by cama36** [comment]
I'm sorry Professor, I'm not sure I'm understanding you correctly. What you mean is that if the point cloud x,y,z ranges from (-10, 10), then 1000*2 voxels are needed to be filled in case the voxel_size is 0.01, and an octree with a depth of 10 is just enough for this resolution, so the depth needs to be greater than 10 in this case (you set it to 12), right?

By the way, my data x,y range in (0,1000), z range in (0,500), voxel_size set to 1.0, depth>=10 is this understanding correct?

**Comment by wang-ps** [comment]
> I'm sorry Professor, I'm not sure I'm understanding you correctly. What you mean is that if the point cloud x,y,z ranges from (-10, 10), then 1000*2 voxels are needed to be filled in case the voxel_size is 0.01, and an octree with a depth of 10 is just enough for this resolution, so the depth needs to be greater than 10 in this case (you set it to 12), right?

Yes

> By the way, my data x,y range in (0,1000), z range in (0,500), voxel_size set to 1.0, depth>=10 is this understanding correct?
Yes
